pylint/lint.py

```diff
@@ -149,7 +149,7 @@ def _patch_sysmodules():
     try:
         yield
     finally:
-        if mock_main:
+        if mock_main and sys.version_info < (3, 3):
```

Comment:
Why do we do this check here? `sys.version_info` cannot be less than 3.3 as pylint does not support older versions. I think we should avoid popping `__main__` altogether.

Reply:
Ohh I think we can remove it entirely then.we only support 3.4+

Reply:
All right, removed it for all versions.